src/workflows/trades.ts

```diff
@@ -24,12 +23,10 @@ export async function createTradeWorkflow(
     signableResult.data;
 
   // Sign message with L1 credentials
-  const ethSignature = await signRaw(signableMessage, signer);
+  const ethSignature = await signRaw(signableMessage, l1Signer);
 
   // Sign hash with L2 credentials
-  const starkSignature = serializeSignature(
-    sign(starkWallet.starkKeyPair, payloadHash),
-  );
+  const starkSignature = await l2Signer.signMessage(payloadHash);
```

Comment:
By looking deeper into how starkWallet signed. it was signed on a converted(fixMessage) **payloadHash**.  
as the fixMessage seems tight to what Starkey generation stuffs. so should we keep the fixMessage in core-sdk.
and change the l2Signer to  sign Message like this:
```suggestion
  const starkSignature = await l2Signer.signMessage(fixMessage(payloadHash));
```

@bruno-imx @zatlite need your input.

Reply:
I would like to know why we need the fixMessage method. As for where it should be implemented I think it should be in the implementation of signMessage instead. 
That's what the kotlin sdk does by the way (see https://github.dev/immutable/imx-core-sdk-kotlin-jvm/blob/96fdbd055c3e555048e938ca689ed5150dcce324/imx-core-sdk-kotlin-jvm/src/main/kotlin/com/immutable/sdk/crypto/StarkKey.kt#L155)

Reply:
that's a good question, I don't think I have the answer, 
@ipekt saw you are the contributor to this code: https://github.com/immutable/imx-core-sdk/blob/main/src/utils/stark/stark-curve.ts#L34 do you have an answer for why we need fixMessage? 

**I think we also missing the implementation in core-sdk**: 
WalletSDK would implement their own signer, But **core-sdk needs the default signer** as well, otherwise, with the changes, there would be no client able to call the workflow without the implementation of **wallet-sdk**. And that's not what we want as we want **core-sdk** to be used in the backend. @zatlite can you confirm the theory. 

In this case, yes, fixMessage can stay in the default signer and walletSDK could potentially extend the default signer to avoid having the fixMessage implemented twice.

Check this POC: https://github.com/immutable/imx-core-sdk/pull/74/files#diff-b0b0da5fd4961644f51fcc6cf05cd5e03373bde43e24b8aa13df7e7c6c039739R73

As for core-sdk default signer implementation, I think we can keep whatever POC have but change the function name like this:

```
export interface StarkSigner {
  signMessage(msg: string): Promise<string>;
  getAddress(): string
}

export class BaseSigner implements StarkSigner {
```

Reply:
Yes I believe that's the plan.

Reply:
Done, added

Reply:
> do you have an answer for why we need fixMessage?

This comes from Stark: 
https://github.com/starkware-libs/starkex-resources/blob/master/crypto/starkware/crypto/signature/signature.js#L274